### DISCUSSION TITLE
Can we use Google cloud storage.

### DISCUSSION THREAD

**Discussion by dildardk123** [Lumieducation/H5P-Nodejs-library#1692] [Q&A]
Hello,

I am using your h5p-package with s3 aws file storage with mongoDb database.
But for the requirement, I want to know can we use google cloud storage with mongodb?
If yes, How will we do this or do you have any other package with google cloud storage?

Thanks

**Comment by sr258** [comment]
I don't see a reason why this shouldn't work, but we don't have a package for this at the moment. You'll have to write storage classes that implements these interfaces:

- ITemporaryFileStorage
- IContentStorage
- possibly, ILibraryStorage, but I'd recommend the MonogLibraryStorage class as object storage is normally too slow for library files

It looks like GCS also supports some kind of compatibility mode for s3: https://cloud.google.com/storage/docs/migrating so you might get away with using the existing Mongo/S3 classes with GCS.
